strategies.py

```diff
@@ -165,9 +165,10 @@ def tree_search(self, num_parallel=None):
             failsafe += 1
             leaf = self.root.select_leaf()
             if self.verbosity >= 4:
-                self.show_path_to_root(leaf)
+                path = self.show_path_to_root(leaf)
+                print(path)
             # if game is over, override the value estimate with the true score
-            if leaf.position.is_game_over():
+            if leaf.position.is_game_over() or leaf.position.n >= MAX_DEPTH:
```

Comment:
Oh man, this is a weird hidden dependency. So, this prevents the MCTS tree search from expanding beyond the max depth, not because it's explicitly forbidden in select_leaf, but because this if clause prevents is_expanded from ever getting set to true. This is some nonintuitive behavior that should probably get called out in a comment somewhere.

Reply:
I don't know if it's that unintuitive -- go.py has the rules of stones on boards.  strategies.py (which should probably be renamed to players.py or something) has all the idiosyncracies about how humans and computers use boards and stones.